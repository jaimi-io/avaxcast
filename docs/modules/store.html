<!DOCTYPE html>
<html class="default no-js">
  <head>
    <meta charset="utf-8" />
    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
    <title>store | avaxcast-frontend</title>
    <meta name="description" content="Documentation for avaxcast-frontend" />
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    <link rel="stylesheet" href="../assets/css/main.css" />
    <script async src="../assets/js/search.js" id="search-script"></script>
  </head>
  <body>
    <header>
      <div class="tsd-page-toolbar">
        <div class="container">
          <div class="table-wrap">
            <div
              class="table-cell"
              id="tsd-search"
              data-index="../assets/js/search.json"
              data-base=".."
            >
              <div class="field">
                <label
                  for="tsd-search-field"
                  class="tsd-widget search no-caption"
                  >Search</label
                >
                <input id="tsd-search-field" type="text" />
              </div>
              <ul class="results">
                <li class="state loading">Preparing search index...</li>
                <li class="state failure">The search index is not available</li>
              </ul>
              <a href="../index.html" class="title">avaxcast-frontend</a>
            </div>
            <div class="table-cell" id="tsd-widgets">
              <div id="tsd-filter">
                <a
                  href="#"
                  class="tsd-widget options no-caption"
                  data-toggle="options"
                  >Options</a
                >
                <div class="tsd-filter-group">
                  <div class="tsd-select" id="tsd-filter-visibility">
                    <span class="tsd-select-label">All</span>
                    <ul class="tsd-select-list">
                      <li data-value="public">Public</li>
                      <li data-value="protected">Public/Protected</li>
                      <li data-value="private" class="selected">All</li>
                    </ul>
                  </div>
                  <input type="checkbox" id="tsd-filter-inherited" checked />
                  <label class="tsd-widget" for="tsd-filter-inherited"
                    >Inherited</label
                  >
                  <input type="checkbox" id="tsd-filter-externals" checked />
                  <label class="tsd-widget" for="tsd-filter-externals"
                    >Externals</label
                  >
                </div>
              </div>
              <a href="#" class="tsd-widget menu no-caption" data-toggle="menu"
                >Menu</a
              >
            </div>
          </div>
        </div>
      </div>
      <div class="tsd-page-title">
        <div class="container">
          <ul class="tsd-breadcrumb">
            <li>
              <a href="../modules.html">avaxcast-frontend</a>
            </li>
            <li>
              <a href="store.html">store</a>
            </li>
          </ul>
          <h1>Module store</h1>
        </div>
      </div>
    </header>
    <div class="container container-main">
      <div class="row">
        <div class="col-8 col-content">
          <section class="tsd-panel-group tsd-index-group">
            <h2>Index</h2>
            <section class="tsd-panel tsd-index-panel">
              <div class="tsd-index-content">
                <section class="tsd-index-section">
                  <h3>Type aliases</h3>
                  <ul class="tsd-index-list">
                    <li class="tsd-kind-type-alias tsd-parent-kind-module">
                      <a href="store.html#AppDispatch" class="tsd-kind-icon"
                        >App<wbr />Dispatch</a
                      >
                    </li>
                    <li class="tsd-kind-type-alias tsd-parent-kind-module">
                      <a href="store.html#RootState" class="tsd-kind-icon"
                        >Root<wbr />State</a
                      >
                    </li>
                  </ul>
                </section>
                <section class="tsd-index-section">
                  <h3>Variables</h3>
                  <ul class="tsd-index-list">
                    <li class="tsd-kind-variable tsd-parent-kind-module">
                      <a href="store.html#default" class="tsd-kind-icon"
                        >default</a
                      >
                    </li>
                    <li class="tsd-kind-variable tsd-parent-kind-module">
                      <a href="store.html#persistor" class="tsd-kind-icon"
                        >persistor</a
                      >
                    </li>
                  </ul>
                </section>
              </div>
            </section>
          </section>
          <section class="tsd-panel-group tsd-member-group">
            <h2>Type aliases</h2>
            <section
              class="
                tsd-panel tsd-member tsd-kind-type-alias tsd-parent-kind-module
              "
            >
              <a name="AppDispatch" class="tsd-anchor"></a>
              <h3>App<wbr />Dispatch</h3>
              <div class="tsd-signature tsd-kind-icon">
                App<wbr />Dispatch<span class="tsd-signature-symbol">:</span>
                <span class="tsd-signature-symbol">typeof </span
                ><span class="tsd-signature-type">store.dispatch</span>
              </div>
              <aside class="tsd-sources">
                <ul>
                  <li>
                    Defined in
                    <a
                      href="https://github.com/jaimi-io/avaxcast/blob/ef67698/frontend/src/store.ts#L10"
                      >src/store.ts:10</a
                    >
                  </li>
                </ul>
              </aside>
            </section>
            <section
              class="
                tsd-panel tsd-member tsd-kind-type-alias tsd-parent-kind-module
              "
            >
              <a name="RootState" class="tsd-anchor"></a>
              <h3>Root<wbr />State</h3>
              <div class="tsd-signature tsd-kind-icon">
                Root<wbr />State<span class="tsd-signature-symbol">:</span>
                <span class="tsd-signature-type">ReturnType</span
                ><span class="tsd-signature-symbol">&lt;</span
                ><span class="tsd-signature-symbol">typeof </span
                ><span class="tsd-signature-type">store.getState</span
                ><span class="tsd-signature-symbol">&gt;</span>
              </div>
              <aside class="tsd-sources">
                <ul>
                  <li>
                    Defined in
                    <a
                      href="https://github.com/jaimi-io/avaxcast/blob/ef67698/frontend/src/store.ts#L9"
                      >src/store.ts:9</a
                    >
                  </li>
                </ul>
              </aside>
            </section>
          </section>
          <section class="tsd-panel-group tsd-member-group">
            <h2>Variables</h2>
            <section
              class="
                tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module
              "
            >
              <a name="default" class="tsd-anchor"></a>
              <h3><span class="tsd-flag ts-flagConst">Const</span> default</h3>
              <div class="tsd-signature tsd-kind-icon">
                default<span class="tsd-signature-symbol">:</span>
                <span class="tsd-signature-type">Store</span
                ><span class="tsd-signature-symbol">&lt;</span
                ><span class="tsd-signature-type">EmptyObject</span
                ><span class="tsd-signature-symbol"> &amp; </span
                ><span class="tsd-signature-symbol">{ </span>isDark<span
                  class="tsd-signature-symbol"
                  >: </span
                ><span class="tsd-signature-type">boolean</span
                ><span class="tsd-signature-symbol">; </span>marketFilter<span
                  class="tsd-signature-symbol"
                  >: </span
                ><a
                  href="../enums/common_enums.Market.html"
                  class="tsd-signature-type"
                  data-tsd-kind="Enumeration"
                  >Market</a
                ><span class="tsd-signature-symbol"> }</span
                ><span class="tsd-signature-symbol"> &amp; </span
                ><span class="tsd-signature-type">PersistPartial</span
                ><span class="tsd-signature-symbol">, </span
                ><a
                  href="../interfaces/lib_global.Action.html"
                  class="tsd-signature-type"
                  data-tsd-kind="Interface"
                  >Action</a
                ><span class="tsd-signature-symbol">&gt;</span
                ><span class="tsd-signature-symbol"> = ...</span>
              </div>
              <aside class="tsd-sources">
                <ul>
                  <li>
                    Defined in
                    <a
                      href="https://github.com/jaimi-io/avaxcast/blob/ef67698/frontend/src/store.ts#L5"
                      >src/store.ts:5</a
                    >
                  </li>
                </ul>
              </aside>
            </section>
            <section
              class="
                tsd-panel tsd-member tsd-kind-variable tsd-parent-kind-module
              "
            >
              <a name="persistor" class="tsd-anchor"></a>
              <h3>
                <span class="tsd-flag ts-flagConst">Const</span> persistor
              </h3>
              <div class="tsd-signature tsd-kind-icon">
                persistor<span class="tsd-signature-symbol">:</span>
                <span class="tsd-signature-type">Persistor</span
                ><span class="tsd-signature-symbol"> = ...</span>
              </div>
              <aside class="tsd-sources">
                <ul>
                  <li>
                    Defined in
                    <a
                      href="https://github.com/jaimi-io/avaxcast/blob/ef67698/frontend/src/store.ts#L6"
                      >src/store.ts:6</a
                    >
                  </li>
                </ul>
              </aside>
            </section>
          </section>
        </div>
        <div class="col-4 col-menu menu-sticky-wrap menu-highlight">
          <nav class="tsd-navigation primary">
            <ul>
              <li class="">
                <a href="../modules.html">Modules</a>
              </li>
              <li class="current tsd-kind-module">
                <a href="store.html">store</a>
              </li>
            </ul>
          </nav>
          <nav class="tsd-navigation secondary menu-sticky">
            <ul class="before-current">
              <li class="tsd-kind-type-alias tsd-parent-kind-module">
                <a href="store.html#AppDispatch" class="tsd-kind-icon"
                  >App<wbr />Dispatch</a
                >
              </li>
              <li class="tsd-kind-type-alias tsd-parent-kind-module">
                <a href="store.html#RootState" class="tsd-kind-icon"
                  >Root<wbr />State</a
                >
              </li>
              <li class="tsd-kind-variable tsd-parent-kind-module">
                <a href="store.html#default" class="tsd-kind-icon">default</a>
              </li>
              <li class="tsd-kind-variable tsd-parent-kind-module">
                <a href="store.html#persistor" class="tsd-kind-icon"
                  >persistor</a
                >
              </li>
            </ul>
          </nav>
        </div>
      </div>
    </div>
    <footer class="with-border-bottom">
      <div class="container">
        <h2>Legend</h2>
        <div class="tsd-legend-group">
          <ul class="tsd-legend">
            <li class="tsd-kind-variable">
              <span class="tsd-kind-icon">Variable</span>
            </li>
            <li class="tsd-kind-function">
              <span class="tsd-kind-icon">Function</span>
            </li>
            <li class="tsd-kind-function tsd-has-type-parameter">
              <span class="tsd-kind-icon">Function with type parameter</span>
            </li>
            <li class="tsd-kind-type-alias">
              <span class="tsd-kind-icon">Type alias</span>
            </li>
          </ul>
          <ul class="tsd-legend">
            <li class="tsd-kind-enum">
              <span class="tsd-kind-icon">Enumeration</span>
            </li>
          </ul>
          <ul class="tsd-legend">
            <li class="tsd-kind-interface">
              <span class="tsd-kind-icon">Interface</span>
            </li>
          </ul>
        </div>
      </div>
    </footer>
    <div class="container tsd-generator">
      <p>
        Generated using
        <a href="https://typedoc.org/" target="_blank">TypeDoc</a>
      </p>
    </div>
    <div class="overlay"></div>
    <script src="../assets/js/main.js"></script>
  </body>
</html>
