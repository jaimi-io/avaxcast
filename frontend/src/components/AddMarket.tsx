import FormControl from "@material-ui/core/FormControl";
import MenuItem from "@material-ui/core/MenuItem";
import Select from "@material-ui/core/Select";
import { createStyles, makeStyles, Theme } from "@material-ui/core/styles";
import TextField from "@material-ui/core/TextField";
import Button from "@material-ui/core/Button";
import PublishIcon from "@material-ui/icons/Publish";
import { useEffect, useState } from "react";
import { getDate } from "common/date";
import Prediction from "contracts/PredictionMarket.json";
import { AbiItem } from "web3-utils";
import { useWeb3React } from "@web3-react/core";
import { menuMarkets } from "common/markets";
import { FLOAT_TO_SOL_NUM, MS_TO_SECS } from "common/constants";
import NumberFormat from "react-number-format";

const MIN_PREDICTED_PRICE = 0;
const ALL_MARKETS_ID = -1;

function validMarket(market: number): boolean {
  return market !== ALL_MARKETS_ID;
}

function validPrice(price: number): boolean {
  return price >= MIN_PREDICTED_PRICE;
}

function validDate(date: string): boolean {
  const now = new Date();
  const selectedDate = new Date(date);
  return now < selectedDate;
}

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: "flex",
      flexWrap: "wrap",
    },
    textField: {
      marginLeft: theme.spacing(1),
      marginRight: theme.spacing(1),
      width: "25ch",
    },
    formControl: {
      margin: theme.spacing(1),
      minWidth: 120,
    },
    container: {
      display: "flex",
      flexWrap: "wrap",
    },
    button: {
      margin: theme.spacing(1),
    },
  })
);

interface NumberFormatCustomProps {
  inputRef: (instance: NumberFormat | null) => void;
  onChange: (event: { target: { name: string; value: string } }) => void;
  name: string;
}

function NumberFormatCustom(props: NumberFormatCustomProps) {
  const { inputRef, onChange, ...other } = props;

  return (
    <NumberFormat
      {...other}
      getInputRef={inputRef}
      onValueChange={(values) => {
        onChange({
          target: {
            name: props.name,
            value: values.value,
          },
        });
      }}
      decimalScale={2}
      thousandSeparator
      fixedDecimalScale
      isNumericString
      prefix="$"
    />
  );
}

function AddMarket(): JSX.Element {
  const classes = useStyles();
  const INITIAL_PREDICTED_PRICE = 0.0;
  const INITIAL_DATE = getDate();
  // state
  const [market, setMarket] = useState(ALL_MARKETS_ID);
  const [predictedPrice, setPredictedPrice] = useState(INITIAL_PREDICTED_PRICE);
  const [deadline, setDeadline] = useState(INITIAL_DATE);
  const [invalid, setInvalid] = useState(true);
  const { active, account, library } = useWeb3React();

  useEffect(() => {
    setInvalid(
      () =>
        !validMarket(market) ||
        !validPrice(predictedPrice) ||
        !validDate(deadline) ||
        !active
    );
  }, [market, predictedPrice, deadline, active]);

  const handleAddMarket = () => {
    const contract = new library.eth.Contract(Prediction.abi as AbiItem[]);
    const unixDeadline = Math.floor(new Date(deadline).getTime() / MS_TO_SECS);
    const marketContract = contract
      .deploy({
        data: Prediction.bytecode,
        arguments: [market, predictedPrice * FLOAT_TO_SOL_NUM, unixDeadline],
      })
      .send({ from: account });
    console.log(marketContract);
  };

  return (
    <div className={classes.root}>
      <FormControl className={classes.formControl}>
        <Select
          defaultValue={ALL_MARKETS_ID}
          labelId="markets-select"
          id="markets-select"
          type="text"
          error={!validMarket(market)}
          // eslint-disable-next-line @typescript-eslint/ban-ts-comment
          // @ts-ignore
          onChange={(e) => setMarket(e.target.value)}>
          <MenuItem value={ALL_MARKETS_ID}>All Markets</MenuItem>
          {menuMarkets().map((mk, index) => (
            <MenuItem value={index} key={index}>
              {mk}
            </MenuItem>
          ))}
        </Select>
      </FormControl>

      <TextField
        label="Predicted Price ($)"
        id="formatted-numberformat-input"
        defaultValue={INITIAL_PREDICTED_PRICE}
        className={classes.textField}
        onChange={(e) => setPredictedPrice(parseFloat(e.target.value))}
        error={predictedPrice < MIN_PREDICTED_PRICE}
        helperText="Range $0 - $X"
        InputProps={{
          // eslint-disable-next-line @typescript-eslint/no-explicit-any
          inputComponent: NumberFormatCustom as any,
        }}
      />

      <form className={classes.container}>
        <TextField
          id="date"
          label="Deadline"
          type="date"
          defaultValue={INITIAL_DATE}
          className={classes.textField}
          InputLabelProps={{
            shrink: true,
          }}
          onChange={(e) => setDeadline(e.target.value)}
          error={!validDate(deadline)}
          helperText={
            validDate(deadline) ? "" : "Select a date after the current date"
          }
        />
      </form>

      <Button
        variant="contained"
        color="inherit"
        className={classes.button}
        disabled={invalid}
        startIcon={<PublishIcon />}
        onClick={() => handleAddMarket()}>
        Submit
      </Button>
    </div>
  );
}

export default AddMarket;
