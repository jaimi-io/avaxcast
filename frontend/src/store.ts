import { createStore } from "redux";
import { persistStore } from "redux-persist";
import rootReducer from "reducers/";

const store = createStore(rootReducer);
export const persistor = persistStore(store);

// Infer the `RootState` and `AppDispatch` types from the store itself
export type RootState = ReturnType<typeof store.getState>;
export type AppDispatch = typeof store.dispatch;
export default store;
