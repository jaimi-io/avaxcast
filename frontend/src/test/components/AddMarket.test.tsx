import renderer from "react-test-renderer";
import AddMarket from "components/AddMarket";
import { Provider } from "react-redux";
import store from "store";
import { BrowserRouter as Router } from "react-router-dom";

test("AddMarket snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Router>
          <AddMarket />
        </Router>
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
