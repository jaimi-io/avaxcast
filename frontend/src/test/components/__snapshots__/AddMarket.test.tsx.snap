// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMarket snapshot 1`] = `
Array [
  <div
    className="MuiGrid-root makeStyles-root-1 MuiGrid-container MuiGrid-justify-content-xs-center"
  >
    <div
      className="MuiFormControl-root makeStyles-formControl-3"
    >
      <div
        className="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-formControl"
        onClick={[Function]}
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="markets-select markets-select"
          className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
          id="markets-select"
          onBlur={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseDown={[Function]}
          role="button"
          tabIndex={0}
        >
          ALL MARKETS
        </div>
        <input
          aria-hidden={true}
          className="MuiSelect-nativeInput"
          onAnimationStart={[Function]}
          onChange={[Function]}
          required={false}
          tabIndex={-1}
          value={4}
        />
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
          style={
            Object {
              "paddingLeft": 8,
            }
          }
        >
          <legend
            className="PrivateNotchedOutline-legend-6"
            style={
              Object {
                "width": 0.01,
              }
            }
          >
            <span
              dangerouslySetInnerHTML={
                Object {
                  "__html": "&#8203;",
                }
              }
            />
          </legend>
        </fieldset>
      </div>
    </div>
    <div
      className="MuiFormControl-root MuiTextField-root makeStyles-textField-2"
    >
      <label
        className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
        data-shrink={false}
        htmlFor="formatted-numberformat-input"
        id="formatted-numberformat-input-label"
      >
        Predicted Price ($)
      </label>
      <div
        className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
        onClick={[Function]}
      >
        <input
          aria-describedby="formatted-numberformat-input-helper-text"
          aria-invalid={false}
          autoFocus={false}
          className="MuiInputBase-input MuiOutlinedInput-input"
          disabled={false}
          id="formatted-numberformat-input"
          inputMode="numeric"
          onAnimationStart={[Function]}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onMouseUp={[Function]}
          required={false}
          type="text"
          value="$0.00"
        />
        <fieldset
          aria-hidden={true}
          className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
        >
          <legend
            className="PrivateNotchedOutline-legendLabelled-7"
          >
            <span>
              Predicted Price ($)
            </span>
          </legend>
        </fieldset>
      </div>
      <p
        className="MuiFormHelperText-root MuiFormHelperText-contained"
        id="formatted-numberformat-input-helper-text"
      >
        Range $0 - $X
      </p>
    </div>
    <form>
      <div
        className="MuiFormControl-root MuiTextField-root makeStyles-textField-2"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined Mui-error Mui-error"
          data-shrink={true}
          htmlFor="date"
          id="date-label"
        >
          Deadline
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root Mui-error Mui-error MuiInputBase-formControl"
          onClick={[Function]}
        >
          <input
            aria-describedby="date-helper-text"
            aria-invalid={true}
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input"
            defaultValue="2021-09-10"
            disabled={false}
            id="date"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            required={false}
            type="date"
          />
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-5 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-7 PrivateNotchedOutline-legendNotched-8"
            >
              <span>
                Deadline
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          className="MuiFormHelperText-root MuiFormHelperText-contained Mui-error"
          id="date-helper-text"
        >
          Select a date after the current date
        </p>
      </div>
    </form>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-contained makeStyles-button-4 MuiButton-colorInherit Mui-disabled Mui-disabled"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <span
          className="MuiButton-startIcon MuiButton-iconSizeMedium"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M5 4v2h14V4H5zm0 10h4v6h6v-6h4l-7-7-7 7z"
            />
          </svg>
        </span>
        Submit
      </span>
    </button>
  </div>,
  <div
    aria-hidden={true}
    className="MuiBackdrop-root makeStyles-backdrop-9"
    style={
      Object {
        "opacity": 0,
        "visibility": "hidden",
      }
    }
  >
    <div
      className="MuiCircularProgress-root MuiCircularProgress-indeterminate"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
  </div>,
]
`;
